### ISSUE TITLE
Is is possible to have callbacks when uploading objects?

### ISSUE THREAD

**Issue by treuille** [aio-libs/aiobotocore#584]
Is it currently possible to get progress callbacks when doing a `client.put_object`?

For example, I know that `boto3` has `upload_file` and `upload_fileobj` methods which have a `Callback` keyword argument. Is there any equivalent in `aiobotocore`?

Thank you!

**Comment by jettify** [comment]
Probably not, aiobotocore based on botocore, I do not see such callbacks there.

@terrycain are such callbacks supported in https://github.com/terrycain/aioboto3 ?

**Comment by thehesiod** [comment]
you could use the base `Future.add_done_callback` no?

```python
task = asyncio.ensure_future(s3_client.put_object(...))
task.add_done_callback(callback)
```

**Comment by thehesiod** [comment]
(removed cmt) sorry, didn't realize you wanted progress

**Comment by terrycain** [comment]
Yeah issue is, `upload_file*` and `download_file*` both essentially call the `s3_transfer` module which deals with the callbacks. I replaced those methods in `aioboto3` with a sorta hacky streaming read / write using get_object / put_object (nastyness [here](https://github.com/terrycain/aioboto3/blob/master/aioboto3/s3/inject.py))

I've implemented the callback thing when the `download_file*` is called, but not in `upload_file*` as i just wait on `put_obj` and let it do all the work.

Create an issue on aioboto3 and I'll look into it. `put_object` takes a file-like object, if it calls read a bunch of times (not just once), callback woudln't be too hard, else might look into doing multipart upload
